api/src/main/java/org/openmrs/Patient.java

```diff
@@ -99,6 +156,9 @@ public Patient(Patient patient) {
 	 * @return internal identifier for patient
 	 */
 	public Integer getPatientId() {
+		if (this.patientId == null) {
```

Comment:
Do you mind explaining why you needed to do this?

Reply:
In-memory, this field is null when you create the patient record since it is technically mapped to a FK column which is why it is mapped as not insertable and not updatable, I wonder why hibernate was setting it with xml but not with annotations though, my guess is there could be a hibernate specific annotations to make fix that JPA doesn't support, will look into it a little a bit to see if there is another work around

Reply:
What do you think of adding this comment to the code?

Reply:
I can but I see that you already merged the PR

Reply:
You can do it as a followup commit.